src/app/components/dungeon/dungeon.component.ts

```diff
@@ -0,0 +1,36 @@
+import { DungeonService } from '../../services/dungeon-service/dungeon.service';
+import { Component, OnInit } from '@angular/core';
+import { Dungeon } from 'src/app/models/dungeons';
+import { ActivatedRoute } from '@angular/router';
+import { Obstacle } from 'src/app/models/obstacle';
+
+@Component({
+  selector: 'app-dungeon',
+  templateUrl: './dungeon.component.html',
+  styleUrls: ['./dungeon.component.scss']
+})
+export class DungeonComponent implements OnInit {
+
+  private _dungeon: Dungeon;
+  private _heroId: number;
+  private _selectedObstacle: Obstacle;
+  private _events: string[];
+
+  constructor(private dungeonService: DungeonService, private _route: ActivatedRoute) {
+    this._heroId = Number(this._route.snapshot.paramMap.get('id'));
+    this.dungeonService.getDungeon(this._heroId).subscribe(response => {
+      this._dungeon = response;
+    });
+  }
+
+  ngOnInit() {
```

Comment:
if youre not using oninit, dont include it

Reply:
fixed